### ISSUE TITLE
`webdav.put` throws 204 if the file already exists

### ISSUE THREAD

**Issue by adil192** [nextcloud/neon#2532]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

### Steps to reproduce
Use `nextcloud: 8.0.0`
```dart
test('Test webdav.put', () async {
  TestWidgetsFlutterBinding.ensureInitialized();
  HttpOverrides.global = null; // enable http requests in test

  final client = NextcloudClient(
    Uri.parse('https://nc.saber.adil.hanney.org'),
    loginName: 'username',
    appPassword: 'password',
  );

  final path = PathUri.parse('/test.webdav.put.txt');
  print('Deleting $path');
  await client.webdav.delete(path);
  print('Trying to put a new file at $path');
  await client.webdav.put(Uint8List.fromList([1, 2, 3]), path);
  print('Trying to put to an existing file at $path');
  await client.webdav.put(Uint8List.fromList([1, 2, 3, 4, 5]), path);
});
```

### Expected behaviour
204 means this and shouldn't throw an error:
`PUT: The action was successful, and the ETag value contains the entity tag for the new representation of that target resource.`

### Actual behaviour
```console
ahann@fedora:~/Documents/GitHub/saber$ flutter test test/nc_put_test.dart 
00:04 +0: Test webdav.put                                                                                                                                                                              
Deleting /test.webdav.put.txt
Trying to put a new file at /test.webdav.put.txt
Trying to put to an existing file at /test.webdav.put.txt
00:05 +0 -1: Test webdav.put [E]                                                                                                                                                                       
  ClientException: Invalid status code 204, headers: {set-cookie: oc_sessionPassphrase=YOinUZBa4vAcOJUEnabOYaVP5RGdjeErk2mCu9IIvoemvtLjCGqMwoUQxNQqaj6lks3En2R8svrFg0teoAK3hXlB5DBJ6yx24Z83mLrLktgQ2TOv4CD6gz0JFet3zyj7; path=/; secure; HttpOnly; SameSite=Lax,__Host-nc_sameSiteCookielax=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=lax,__Host-nc_sameSiteCookiestrict=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=strict,oc5clkm8tw2o=akrae8et0afcqcbusv1iba7u3m; path=/; secure; HttpOnly; SameSite=Lax, date: Tue, 01 Oct 2024 02:58:05 GMT, strict-transport-security: max-age=63072000; includeSubdomains, referrer-policy: no-referrer, x-robots-tag: noindex, nofollow, x-permitted-cross-domain-policies: none, x-xss-protection: 1; mode=block, server: Apache, x-request-id: JYS49D9Rv94JJcbSPDvp, oc-etag: "605e6524c234ec1130affa0e3ff1ad32", oc-fileid: 01533865oc5clkm8tw2o, etag: "605e6524c234ec1130affa0e3ff1ad32", x-frame-options: SAMEORIGIN, x-content-type-options: nosniff, content-security-policy: default-src 'none';}, body: , uri=https://nc.saber.adil.hanney.org/remote.php/webdav/test.webdav.put.txt
  package:nextcloud/src/api/webdav/webdav_client.dart 216:7  WebDavClient.put
  

To run this test again: /home/ahann/Documents/Sources/flutter/bin/cache/dart-sdk/bin/dart test /home/ahann/Documents/GitHub/saber/test/nc_put_test.dart -p vm --plain-name 'Test webdav.put'
00:05 +0 -1: Some tests failed.                  
```

### Server configuration

**Web server:** Apache

**Database:** MySQL

**PHP version:** 8.2.22

**Nextcloud version:** Nextcloud Hub 8 (29.0.7)

<details>
<summary>List of activated apps</summary>

```console
# nextcloud.occ app:list
Enabled:
  - activity: 2.21.1
  - admin_audit: 1.19.0
  - bruteforcesettings: 2.9.0
  - cloud_federation_api: 1.12.0
  - dav: 1.30.1
  - drop_account: 2.6.0
  - federatedfilesharing: 1.19.0
  - files: 2.1.1
  - files_pdfviewer: 2.10.0
  - files_versions: 1.22.0
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - oauth2: 1.17.1
  - password_policy: 1.19.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - registration: 2.5.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - suspicious_login: 7.0.0
  - theming: 2.4.0
  - twofactor_admin: 4.5.0
  - twofactor_backupcodes: 1.18.0
  - twofactor_nextcloud_notification: 3.10.0
  - twofactor_totp: 11.0.0-dev
  - twofactor_webauthn: 1.4.0
  - viewer: 2.3.0
  - workflowengine: 2.11.0
Disabled:
  - circles: 29.0.0-dev (installed 27.0.1)
  - comments: 1.19.0 (installed 1.17.0)
  - contactsinteraction: 1.10.0 (installed 1.6.0)
  - dashboard: 7.9.0 (installed 7.7.0)
  - encryption: 2.17.0 (installed 2.17.0)
  - federation: 1.19.0 (installed 1.17.0)
  - files_downloadlimit: 2.0.0 (installed 2.0.0)
  - files_external: 1.21.0 (installed 1.17.0)
  - files_reminders: 1.2.0 (installed 1.2.0)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - files_sharing: 1.21.0 (installed 1.19.0)
  - files_trashbin: 1.19.0 (installed 1.17.0)
  - firstrunwizard: 2.18.0 (installed 2.14.0)
  - nextcloud_announcements: 1.18.0 (installed 1.14.0)
  - notifications: 2.17.0 (installed 2.15.0)
  - photos: 2.5.0 (installed 2.3.0)
  - recommendations: 2.1.0 (installed 1.6.0)
  - related_resources: 1.4.0 (installed 1.0.0)
  - sharebymail: 1.19.0 (installed 1.17.0)
  - support: 1.12.0 (installed 1.10.0)
  - survey_client: 1.17.0 (installed 1.12.0)
  - systemtags: 1.19.0 (installed 1.17.0)
  - text: 3.10.1 (installed 3.8.0)
  - twofactor_email: 2.7.4 (installed 2.7.4)
  - user_ldap: 1.20.0 (installed 1.17.0)
  - user_status: 1.9.0 (installed 1.4.0)
  - weather_status: 1.9.0 (installed 1.4.0)
```
</details>

<details>
<summary>Nextcloud configuration</summary>

```console
# nextcloud.occ config:list system
{
    "system": {
        "apps_paths": [
            {
                "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
                "url": "\/extra-apps",
                "writable": true
            }
        ],
        "supportedDatabases": [
            "mysql"
        ],
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "log_type": "file",
        "logfile": "\/var\/snap\/nextcloud\/current\/logs\/nextcloud.log",
        "logfilemode": 416,
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nc.saber.adil.hanney.org",
            "www.nc.saber.adil.hanney.org"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "29.0.7.1",
        "overwrite.cli.url": "https:\/\/nc.saber.adil.hanney.org\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "GB",
        "profile.enabled": false,
        "skeletondirectory": "",
        "maintenance": false,
        "data-fingerprint": "626b068304e67d37347be482fad577b6",
        "mail_smtpmode": "smtp",
        "mail_smtpauth": 1,
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 1,
        "theme": "",
        "maintenance_window_start": 1,
        "app_install_overwrite": [
            "twofactor_email"
        ],
        "versions_retention_obligation": "auto, 7"
    }
}
```
</details>

**Comment by provokateurin** [comment]
Thanks for reporting. The correct reference would be https://datatracker.ietf.org/doc/html/rfc4918#section-9.7.1 which explains the content is silently replaced (but doesn't mention any status code).

**Comment by provokateurin** [comment]
Released in 8.0.1
